sqlmesh/core/engine_adapter/clickhouse.py

```diff
@@ -174,6 +175,198 @@ def create_schema(
             kind="DATABASE",
         )
 
+    def _insert_overwrite_by_condition(
+        self,
+        table_name: TableName,
+        source_queries: t.List[SourceQuery],
+        columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None,
+        where: t.Optional[exp.Condition] = None,
+        insert_overwrite_strategy_override: t.Optional[InsertOverwriteStrategy] = None,
+        **kwargs: t.Any,
+    ) -> None:
+        target_table = exp.to_table(table_name)
+        columns_to_types = columns_to_types or self.columns(target_table)
+
+        temp_table = self._get_temp_table(target_table)
+        self._create_table_like(temp_table, target_table)
```

Comment:
you're doing a full copy?

Reply:
No, this creates an empty table with the same structure (columns, data types, partitioning, etc.) as target